### ISSUE TITLE
Restoring library never completes

### ISSUE THREAD

**Issue by ManuelHaas** [aspnet/LibraryManager#512]
### Functional impact
Restore never finishes

### Minimal repro steps
Try this config with VS 16.3.0:

{
  "version": "1.0",
  "defaultProvider": "cdnjs",
  "libraries": [
    {
      "library": "metisMenu@3.0.4",
      "destination": "wwwroot/lib/metisMenu/"
    },
    {
      "provider": "unpkg",
      "library": "devextreme@19.1.5",
      "destination": "wwwroot/lib/devextreme/"
    }
  ]
}

I already tried to clean the libs before restore, restarted VS and tried again. Always the same - it never ends.

### Actual result
Restore never finishes. Stopped restore after 1200 seconds. 
First libarary was restored, second not.
Not a single file was in the directory afterwards!

### Further technical details
Here is the log of stopping the restore after 20! minutes:

Restore operation started...
Restoring libraries for project ToSec.Presentation
Restoring library metisMenu@3.0.4... (ToSec.Presentation)
wwwroot/lib/metisMenu/metisMenu.css written to destination (ToSec.Presentation)
wwwroot/lib/metisMenu/metisMenu.css.map written to destination (ToSec.Presentation)
wwwroot/lib/metisMenu/metisMenu.js written to destination (ToSec.Presentation)
wwwroot/lib/metisMenu/metisMenu.js.map written to destination (ToSec.Presentation)
wwwroot/lib/metisMenu/metisMenu.min.css written to destination (ToSec.Presentation)
wwwroot/lib/metisMenu/metisMenu.min.css.map written to destination (ToSec.Presentation)
wwwroot/lib/metisMenu/metisMenu.min.js written to destination (ToSec.Presentation)
wwwroot/lib/metisMenu/metisMenu.min.js.map written to destination (ToSec.Presentation)
Restoring library devextreme@19.1.5... (ToSec.Presentation)
Restore operation completed
1 libraries restored in 1219,11 seconds
1 libraries failed to restore

**Comment by ddweber** [comment]
I can confirm this issue. It seems as if UNPKG as library provider is the cause of this problem.
The GUI of Visual Studio sometimes even freezed/crashed when i tried searching for a lib on UNPKG via LibMan.

**Comment by ddweber** [comment]
I tested the same libman.json via the CLI and it worked. It took ages but it executed without any errors.

**Comment by dmitry-pavlov** [comment]
I confirm that LibMan restore never ends or takes ages 😭 with any provider - #cdnjs, #unpkg, doesn't matter. I would love to use LibMan for client JS libs, but restoring just doesn't seems to complete even withing a week... neither via VS UI nor via CLI. Something is definitely wrong with it in the latest Visual Studio Community 2019 - Version 16.3.0 Preview 2.0.

**Comment by wgutierrezr** [comment]
Any news about this issue? Please help.

**Comment by jimmylewis** [comment]
I tried this out with the original example (devextreme, which contains 1000+ files).  It took ~75 minutes to download these files (that's about 4 sec per file, in series), and no intermediate logging.  I verified that it was still going by monitoring network traffic and also attaching the debugger (a perf trace wouldn't have shown much, as it's all in the same loop).

Obviously, there's some **significant** room for improvements here.  However, I haven't seen a case where it *never* ends (either successfully or with an error).  Has anyone here encountered an actual hang?
